source/text_handler.cpp

```diff
@@ -234,22 +236,11 @@ bool AssemblyContext::hasText() const {
 }
 
 std::string AssemblyContext::getWord() const {
-  uint64_t index = current_position_.index;
-  while (true) {
-    switch (text_->str[index]) {
-      case '\0':
-      case '\t':
-      case '\v':
-      case '\r':
-      case '\n':
-      case ' ':
-        return std::string(text_->str, text_->str + index);
```

Comment:
This is the direct reason why #126 behaves like that. We should return `string(text_->str + current_position_.index, ...)`.